qcore/validate_vm.py

```diff
@@ -213,26 +216,26 @@ def validate_vm_bounds(polygon, srf_bounds=None):
     return errors
 
 
-def validate_vm_file(file_name: Path, vm_size: int):
+def validate_vm_file(file_name: Path, nx: int, ny: int, nz: int):
     """
-    Validates that a velocity model file has the correct size, and no 0 values
+    Validates that a velocity model file has the correct size, and no 0 values in a sample of the layers
     :param file_name: A Path object representing the file to test
-    :param vm_size: The size of the VM in grid spaces (nx*ny*nz)
+    :param nx, ny, nz: The size of the VM in grid spaces (nx*ny*nz)
     :return: A possibly empty list of issues with the VM file
     """
     errors = []
+    vm_size = nx * ny * nz
     size = file_name.stat().st_size
     if size != vm_size * SIZE_FLOAT:
         errors.append(
             f"VM filesize for {file_name} expected: {vm_size * SIZE_FLOAT} found: {size}"
         )
-    if (
-        not np.min(
-            np.fromfile(file_name, dtype="<f{}".format(SIZE_FLOAT), count=vm_size)
-        )
-        > 0
-    ):
-        errors.append(f"File {file_name} has minimum value of 0.0")
+
+    with VelocityModelFile(nx, ny, nz, file_name, memmap=True) as vmf:
```

Comment:
just to be safe, probably add a
writable=False
into this?

Reply:
its the default behavior currently, but since we are sure we only need it to be read-only, adding the flag wont hurt.